core/embed/trezorhal/stm32u5/lowlevel.c

```diff
@@ -129,9 +130,8 @@ secbool flash_check_option_bytes(void) {
   }
 
 #if PRODUCTION
-  // TODO error this wont work, need to add default/reset values
-#error this wont work, need to add default/reset values
-  if (FLASH->WRP1AR != (WANT_WRP_PAGE_START | (WANT_WRP_PAGE_END << 16))) {
+  if (FLASH->WRP1AR !=
+      (WRP_LOCKED_VALUE | WANT_WRP_PAGE_START | (WANT_WRP_PAGE_END << 16))) {
```

Comment:
It would be better to have two constants with a little bit different purpose:

```c
#define WRPA1R_DEFAULT_VALUE 0xFF80FFFF  -- this is MCU dependent
#define WRP1AR_LOCKED_VALUE  ((WRP1AR_DEFAULT_VALUE & ...) | ...)
```
This allows comparison and potential setting of the register to the same constant without redundancy.

Reply:
https://github.com/trezor/trezor-firmware/commit/9e229edbfb823c15ce42616a626879327154ce65